### PR TITLE
Get walk from hash in url

### PR DESCRIPTION
**Pull request by shearnie**
On open, select the walk based on the hash value in the url, so when a person clicks on the share link in Facebook it will show the walk being shared.

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -222,9 +222,13 @@ function setupRoutes(map, pos, targetRouteLength) {
         pos.lat + KM_RADIUS_TO_SEARCH_FOR_ROUTES_NEAR,
         pos.lng + KM_RADIUS_TO_SEARCH_FOR_ROUTES_NEAR,
         function(data) {
-            var walkRoute = getRandomWalkFromOsmDataset(data, targetRouteLength);
-
-            renderWalk(walkRoute, pos);
+            if (hashVal) {
+                var walkRoute = getWalkFromOsmDatasetById(data, hashVal.split('#')[1]);
```
Comment (Novex) [inline assets/js/app.js]: Hmm.. I can see a problem here in that if you aren't located in the same radius as your friend the walk id may not be in the OSM dataset - eg. if I'm on the gold coast and you send me a walk in brisbane that's not within 20km of me I won't see it :(

We might have to query OSM directly for the specific walk id using a different api query to make sure we get the right id

File: assets/js/osm.js [hunk 2]
```diff
@@ -71,6 +71,17 @@ function addNodeInformation(data) {
     return data;
 }
 
+function getWalkFromOsmDatasetById(data, id) {
+  var walks = [];
```
Comment (Novex) [inline assets/js/osm.js]: unused? :)

**Review by Novex — COMMENTED** [review]
looks great! just one thing to take a look at

**Comment by shearnie** [comment]
@Novex this looks ready to merge. Let me know if any more issues.

File: assets/js/osm.js [hunk 3]
```diff
@@ -29,6 +29,19 @@ function getOsmNodes(slat, slng, nlat, nlng, callback) {
     });
 }
 
+function getOsmWay(id, callback) {
+    var osmQuery = '[out:json];way(' + id + ')[highway=path][foot=yes][surface=ground];out body;>;out skel qt;';
```
Comment (Novex) [inline assets/js/osm.js]: Does the `[highway=path][foot=yes][surface=ground]` restrict the way id's to only those attributes? If we dropped that part of the query in here and just loaded any old way id then we wouldn't have to keep this in sync with the query in `getOsmNodes` when it changes :)

Comment (shearnie) [reply assets/js/osm.js]: Doesn't restrict, but I can remove them anyway for peace of mind.

**Pull request merged**